lib/ipam/ipam.go

```diff
@@ -446,6 +437,9 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 						logCtx.WithError(err).Warningf("Failed to assign IPs from non-affine block in pool %s", p.String())
 						break
 					}
+					if len(newIPs) == 0 {
+						break
```

Comment:
Would it be worth adding a log line here?

Reply:
I had one but removed it because we already generate a log line when there are no IPs.